Apps: Add clean scripts

sirreal opened the pull request:
Adds `clean` scripts following conventions from other packages and apps in the monorepo.

#### Changes proposed in this Pull Request

* All the apps now have clean scripts and follow a convention.

#### Testing instructions

* `npx lerna run clean` should result in apps being cleaned up
* Clean scripts should run before relevant scripts, like build and dev.
* Artifacts produced by relevant CircleCI jobs should be unchanged.

Noticed in #36583

sirreal committed:
Add conventional clean scripts to apps

simison reviewed (approved):
Looks good and CI passes.

"Owner" teams could still give thumbsup before merging tho.

simison commented on apps/full-site-editing/package.json:
```diff
@@ -25,11 +25,14 @@
-		"dev": "rimraf dist && npm-run-all --parallel dev:*",
-		"build": "rimraf dist && npm-run-all --parallel build:*",
+		"dev": "npm-run-all --parallel dev:*",
+		"build": "npm-run-all --parallel build:*",
```
I really appreciate this long form instead of `run-p` 👍  ...should do the same in the root `package.json` :-)

vindl reviewed (approved):
Thanks for adding this! LGTM! :shipit:

One thing that I've noticed is that Artifacts tab (where you could browse the generated folder structure) is no longer present for these jobs, but that seems to be a recent change on the CircleCI side? In any case I verified that our sync scripts can still fetch them when passed the the appropriate job ID.

mmtr commented:
> One thing that I've noticed is that Artifacts tab (where you could browse the generated folder structure) is no longer present for these jobs, but that seems to be a recent change on the CircleCI side? In any case I verified that our sync scripts can still fetch them when passed the the appropriate job ID.

That happened to me in the past when I was logged out of CircleCI. Make sure you're authenticated with GitHub (https://circleci.com/vcs-authorize/) and you should see the Artifacts tab again.

vindl commented:
> Make sure you're authenticated with GitHub (https://circleci.com/vcs-authorize/) and you should see the Artifacts tab again.

Yup, that was it. Thanks!

One more note regarding the PR - this doesn't seem to be deleting `dist` folders in all of the appropriate FSE places. Due to placing multiple projects into a single plugin, every one of them has a separate `dist` folder (Full Site Editing, Starter Page Templates, Posts List Block). These were supposed to be split out into separate plugins, but the work on that has been punted recently.

sirreal commented on apps/full-site-editing/package.json:
```diff
@@ -25,11 +25,14 @@
 		"test:unit:help": "wp-scripts test-unit-js --config='jest.config.js' --help",
-		"test:unit:watch": "wp-scripts test-unit-js --config='jest.config.js' --watch"
+		"test:unit:watch": "wp-scripts test-unit-js --config='jest.config.js' --watch",
+		"clean": "npx rimraf dist",
```
It sounds like this clean is incomplete and a couple more directories should be removed.

Via @vindl https://github.com/Automattic/wp-calypso/pull/36602#issuecomment-539942350

sirreal replied:
I tracked these down and added them to clean in 6376c95f839f39c8f399ec07ec1b774ad2d2c2ca

sirreal committed:
Clean fse-plugin dist directories

sirreal committed:
Use custom gitignore for full-site-editing

sirreal commented:
Rebased.

I added additional directories to clean and adjusted the way we handle ignores (standard `dist` directory ignored globallay, custom `dist` directories ignored in the app).

This should be ready to merge. If anyone has feedback, let me know. Otherwise I'll plan to merge early next week.
